lang/csharp/src/apache/main/File/DataFileReader.cs

```diff
@@ -245,6 +233,8 @@ public bool PastSync(long position)
         /// <inheritdoc/>
         public long PreviousSync()
         {
+            if (!_stream.CanSeek)
+                throw new AvroException("Not a valid input stream - must be seekable!");
```

Comment:
Throw an Avro**Runtime**Exception here instead.

```suggestion
                throw new AvroRuntimeException("Not a valid input stream - must be seekable!");
```